Code/Tools/ProjectManager/Source/CreateAGemScreen.cpp

```diff
@@ -0,0 +1,446 @@
+#include <PythonBindingsInterface.h>
+
+
+#include <PythonBindingsInterface.h>
+#include <QDir>
+#include <QFileDialog>
+#include <QHBoxLayout>
+#include <QHash>
+#include <QMessageBox>
+#include <QPushButton>
+#include <QStackedWidget>
+#include <QStandardPaths>
+#include <QVBoxLayout>
+#include <QRadioButton>
+#include <QLabel>
+#include <QDialogButtonBox>
+
+
+namespace O3DE::ProjectManager
+{
+    CreateAGemScreen::CreateAGemScreen(QWidget* parent)
+        : ScreenWidget(parent)
+    {
+        QVBoxLayout* vLayout = new QVBoxLayout();
+        vLayout->setSpacing(0);
+        vLayout->setContentsMargins(0, 0, 0, 0);
+
+        QScrollArea* firstScreenScrollArea = CreateFirstScreen();
+        QScrollArea* secondScreenScrollArea = CreateSecondScreen();
+        QScrollArea* thirdScreenScrollArea = CreateThirdScreen();
+
+        connect(m_gemDisplayName->lineEdit(), &QLineEdit::textChanged, this, &CreateAGemScreen::OnGemDisplayNameUpdated);
+        connect(m_gemSystemName->lineEdit(), &QLineEdit::textChanged, this, &CreateAGemScreen::OnGemSystemNameUpdated);
+        connect(m_license->lineEdit(), &QLineEdit::textChanged, this, &CreateAGemScreen::OnLicenseNameUpdated);
+        connect(m_creatorName->lineEdit(), &QLineEdit::textChanged, this, &CreateAGemScreen::OnCreatorNameUpdated);
+        connect(m_repositoryURL->lineEdit(), &QLineEdit::textChanged, this, &CreateAGemScreen::OnRepositoryURLUpdated);
+
+        m_tabWidget = new TabWidget();
+        m_tabWidget->setObjectName("createAGemTab");
+        m_tabWidget->addTab(firstScreenScrollArea, "1. Gem Setup");
+        m_tabWidget->addTab(secondScreenScrollArea, "2. Gem Details");
+        m_tabWidget->addTab(thirdScreenScrollArea, "3. Creator Details");
+        m_tabWidget->tabBar()->setEnabled(false);
+
+        
+        vLayout->addWidget(m_tabWidget);
+
+        QFrame* footerFrame = new QFrame();
+        footerFrame->setObjectName("createAGemFooter");
+        m_backNextButtons = new QDialogButtonBox();
+        m_backNextButtons->setObjectName("footer");
+        QVBoxLayout* footerLayout = new QVBoxLayout();
+        footerLayout->setContentsMargins(0, 0, 0, 0);
+        footerFrame->setLayout(footerLayout);
+        footerLayout->addWidget(m_backNextButtons);
+        vLayout->addWidget(footerFrame);
+
+
+        m_backButton = m_backNextButtons->addButton(tr("Back"), QDialogButtonBox::RejectRole);
+        m_backButton->setProperty("secondary", true);
+        m_nextButton = m_backNextButtons->addButton(tr("Next"), QDialogButtonBox::ApplyRole);
+
+        connect(m_tabWidget, &TabWidget::currentChanged, this, &CreateAGemScreen::ConvertNextButtonToCreate);
+        connect(m_backButton, &QPushButton::clicked, this, &CreateAGemScreen::HandleBackButton);
+        connect(m_nextButton, &QPushButton::clicked, this, &CreateAGemScreen::HandleNextButton);
+
+        connect(m_nextButton, &QPushButton::clicked, this, &CreateAGemScreen::OnGemDisplayNameUpdated);
```

Comment:
We could have this as part of the next page logic since we need to validate there to be able to progress to the next page anyway, saves having all of these connect calls too.

Reply:
Added to HandleNextPage()